modules/Evse15118D20/charger/ISO15118_chargerImpl.cpp

```diff
@@ -112,6 +112,108 @@ auto fill_mobility_needs_modes_from_config(const module::Conf& module_config) {
+    // Fill in the common data
+    fill_v2x_charging_parameters<iso15118::d20::DcTransferLimits, dt::DC_CPDReqEnergyTransferMode>(
+        out_params, evse_limits, static_cast<const dt::DC_CPDReqEnergyTransferMode&>(ev_limits));
+
+    // Fill in the bidi data
+    if (evse_limits.discharge_limits.has_value()) {
+        const auto& evse_discharge_limits = evse_limits.discharge_limits.value();
+
+        out_params.min_discharge_power = std::max(dt::from_RationalNumber(evse_discharge_limits.power.min),
+                                                  dt::from_RationalNumber(ev_limits.min_discharge_power));
+        out_params.max_discharge_power = std::min(dt::from_RationalNumber(evse_discharge_limits.power.max),
+                                                  dt::from_RationalNumber(ev_limits.max_discharge_power));
+
+        out_params.min_discharge_current = std::max(dt::from_RationalNumber(evse_discharge_limits.current.min),
+                                                    dt::from_RationalNumber(ev_limits.min_discharge_current));
+        out_params.max_discharge_current = std::min(dt::from_RationalNumber(evse_discharge_limits.current.max),
+                                                    dt::from_RationalNumber(ev_limits.max_discharge_current));
+    }
+}
+
+template <>
+void fill_v2x_charging_parameters(types::iso15118::V2XChargingParameters& out_params,
+                                  const dt::Scheduled_SEReqControlMode& ev_control_mode) {
+    out_params.ev_target_energy_request = convert_from_optional<float>(ev_control_mode.target_energy);
+    out_params.ev_min_energy_request = convert_from_optional<float>(ev_control_mode.min_energy);
+    out_params.ev_max_energy_request = convert_from_optional<float>(ev_control_mode.max_energy);
+}
+
+template <>
+void fill_v2x_charging_parameters(types::iso15118::V2XChargingParameters& out_params,
+                                  const dt::Dynamic_SEReqControlMode& ev_control_mode) {
+    out_params.ev_target_energy_request = dt::from_RationalNumber(ev_control_mode.target_energy);
+    out_params.ev_min_energy_request = dt::from_RationalNumber(ev_control_mode.min_energy);
+    out_params.ev_max_energy_request = dt::from_RationalNumber(ev_control_mode.max_energy);
+
+    out_params.ev_min_v2xenergy_request = convert_from_optional<float>(ev_control_mode.min_v2x_energy);
+    out_params.ev_max_v2xenergy_request = convert_from_optional<float>(ev_control_mode.max_v2x_energy);
+}
+
+types::iso15118::EnergyTransferMode get_energy_transfer_mode(const dt::ServiceCategory& service_category,
+                                                             const std::optional<dt::AcConnector>& ac_connector) {
+    using namespace types::iso15118;
+
+    EnergyTransferMode requested_energy_transfer = EnergyTransferMode::AC_single_phase_core;
+
+    if (service_category == dt::ServiceCategory::AC) {
+        if (ac_connector.has_value()) {
```

Comment:
Combine this both conditions in one `if (service_category == dt::ServiceCategory::AC and ac_connector.has_value()) {`